### DISCUSSION TITLE
Extract all bitmaps from a pdf.

### DISCUSSION THREAD

**Discussion by granttonya** [ImageMagick/ImageMagick#2648] [Help]
I would like to use magick commend line to extract all bmp files in a pdf document into a directory on windows.
Thanks for any help.

**Comment by snibgo** [comment]
I suggest pdfimages, not ImageMagick, for that.

IM won't extract raster images (of which there may be many per PDF page, or none at all). Instead, it will (via Ghostscript) rasterize each PDF page into a raster image, one image per page.

**Reply by granttonya** [reply]
Thanks for the help :)
